lisa/sut_orchestrator/azure/features.py

```diff
@@ -1134,61 +1134,80 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
         self._initialize_information(self._node)
 
     def get_raw_data_disks(self) -> List[str]:
-        # refer here to get data disks from folder /dev/disk/azure/scsi1
-        # https://docs.microsoft.com/en-us/troubleshoot/azure/virtual-machines/troubleshoot-device-names-problems#identify-disk-luns  # noqa: E501
-        # /dev/disk/azure/scsi1/lun0
-        ls_tools = self._node.tools[Ls]
-        files = ls_tools.list("/dev/disk/azure/scsi1", sudo=True)
-
-        if len(files) == 0:
-            os = self._node.os
-            # there are known issues on ubuntu 16.04 and rhel 9.0
-            # try to workaround it
-            if (isinstance(os, Ubuntu) and os.information.release <= "16.04") or (
-                isinstance(os, Redhat) and os.information.release >= "9.0"
-            ):
-                self._log.debug(
-                    "download udev rules to construct a set of symbolic links "
-                    "under the /dev/disk/azure path"
-                )
-                if ls_tools.is_file(
-                    self._node.get_pure_path("/dev/disk/azure"), sudo=True
+        disk_array: List[str] = []
+
+        if (
+            self._node.capability.disk
```

Comment:
can we have a new case, and put this logic in case level?

Reply:
We are still testing the same premium disks with disk_controller_type==NVMe. The only difference is that they are exposed to guest as NVMe devices. The reason is to go beyond the limits of SCSI protocol so we are using NVMe protocol/driver instead of SCSI.  So we can still use the same testcase.